apps/extension/src/wallet-services.ts

```diff
@@ -6,21 +6,49 @@ import {
 } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/keys/v1/keys_pb';
 import { onboardGrpcEndpoint, onboardWallet } from './storage/onboard';
 import { ServicesMessage } from './message/services';
+import { localExtStorage } from './storage/local';
+import { AppParameters } from '@buf/penumbra-zone_penumbra.bufbuild_es/penumbra/core/app/v1/app_pb';
+import { AppService } from '@penumbra-zone/protobuf';
+import { createGrpcWebTransport } from '@connectrpc/connect-web';
+import { createPromiseClient } from '@connectrpc/connect';
 
 export const startWalletServices = () =>
   backOff(
     async () => {
       const wallet = await onboardWallet();
       const grpcEndpoint = await onboardGrpcEndpoint();
+
+      const chainId =
+        (await localExtStorage
+          .get('params')
+          .then(json => json && AppParameters.fromJson(json).chainId)) ??
+        (await createPromiseClient(AppService, createGrpcWebTransport({ baseUrl: grpcEndpoint }))
+          .appParameters({})
+          .then(({ appParameters }) => appParameters?.chainId));
+
+      if (!chainId) throw new Error('No chainId available');
+
       const services = new Services({
+        chainId,
         idbVersion: IDB_VERSION,
         grpcEndpoint,
         walletId: WalletId.fromJsonString(wallet.id),
         fullViewingKey: FullViewingKey.fromJsonString(wallet.fullViewingKey),
       });
```

Comment:
chainId is retrieved from existing storage, or queried before launching services-context

it is also stored in idb, but we can't connect to idb without a chainid, as the database name includes chainid.

this could be simplified - we could store chainId as a string, instead of all params, and then we can retrieve params from idb. but that may require a storage migration at this point